### ISSUE TITLE
defineProps is warned by no-undef rule

### ISSUE THREAD

**Issue by sajjad5221** [vuejs/eslint-plugin-vue#1646]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version: 7.32.0 
- **eslint-plugin-vue version: 7.18.0 
- **Node version: 14.17.1
- **Operating System: Ubuntu 20.04

**Please show your full configuration:**

```json5
{
	"root": true,
	"env": {
		"node": true,
		"browser": true
	},
	"parser": "vue-eslint-parser",
	"parserOptions": {
		"parser": "espree",
		"ecmaVersion": 2022,
		"sourceType": "module"
	},
	"extends": [
		"standard",
		"plugin:vue/vue3-recommended"
	],
	"plugins": [
		"import"
	],
	"rules": {
		"quotes": [
			2,
			"single"
		],
		"indent": [
			"error",
			"tab"
		],
		"no-tabs": [
			"error",
			{
				"allowIndentationTabs": true
			}
		],
		"comma-dangle": [
			"error",
			"always-multiline"
		],
		"max-lines": [
			"error",
			{
				"max": 360,
				"skipBlankLines": true,
				"skipComments": true
			}
		],
		"vue/html-indent": [
			"error",
			"tab"
		],
		"vue/attribute-hyphenation": [
			"error",
			"always"
		],
		"vue/component-definition-name-casing": [
			"error",
			"PascalCase"
		],
		"vue/html-self-closing": [
			"error",
			{
				"html": {
					"void": "always",
					"normal": "always",
					"component": "always"
				},
				"svg": "always",
				"math": "always"
			}
		],
		"vue/multiline-html-element-content-newline": [
			"error",
			{
				"ignoreWhenEmpty": true,
				"ignores": [
					"pre",
					"textarea"
				],
				"allowEmptyLines": false
			}
		],
		"vue/html-closing-bracket-newline": [
			"error",
			{
				"singleline": "never",
				"multiline": "always"
			}
		],
		"vue/no-unused-vars": "error",
		"semi": [
			"error",
			"always"
		],
		"import/newline-after-import": [
			"error",
			{
				"count": 1
			}
		],
		"import/imports-first": [
			"error",
			"absolute-first"
		],
		"space-before-function-paren": [
			"error",
			"always"
		],
		"object-curly-spacing": [
			"error",
			"always"
		],
		"no-console": [
			"error",
			{
				"allow": [
					"debug",
					"error"
				]
			}
		],
		"vue/max-attributes-per-line": [
			"error",
			{
				"singleline": {
					"max": 2,
					"allowFirstLine": true
				},
				"multiline": {
					"max": 1,
					"allowFirstLine": false
				}
			}
		],
		"vue/component-name-in-template-casing": [
			"error",
			"PascalCase",
			{
				"registeredComponentsOnly": false
			}
		],
		"vue/custom-event-name-casing": [
			"error",
			"kebab-case"
		],
		"vue/no-empty-component-block": "error",
		"vue/no-useless-mustaches": "error",
		"vue/no-useless-v-bind": "error",
		"vue/no-unused-properties": [
			"error",
			{
				"groups": [
					"props",
					"setup"
				]
			}
		],
		"vue/padding-line-between-blocks": [
			"error",
			"always"
		],
		"vue/script-setup-uses-vars": "error"
		// "vue/valid-define-props":"allow"
	},
	"globals": {
		"tns": "readonly",
		"Swal": "readonly",
		"defineProps": "readonly",
		"defineEmits": "readonly"
	}
}


```

**What did you do?**	

```vue
<template>
	<ElSkeleton animated>
		<template #template>
			<ElSkeletonItem
				:style="{ width: width, height: height }"
				variant="image"
			/>
		</template>
	</ElSkeleton>
</template>

<script setup>

const props = defineProps({
	width: {
		type: Number,
		default: null,
	},
	height: {
		type: Number,
		default: null,
	},
});
```

**What did you expect to happen?**
eslint success ( it's success in some other components)

**What actually happened?**

error  'props' is assigned a value but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)

**Repository to reproduce this issue**

**Comment by ota-meshi** [comment]
I think you can delete the props variable.

```js
defineProps({
	width: {
		type: Number,
		default: null,
	},
	height: {
		type: Number,
		default: null,
	},
});
```

It works with [DEMO](https://sfc.vuejs.org/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBZb3VyQ29tcG9uZW50IGZyb20gJy4vWW91ckNvbXBvbmVudC52dWUnXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8WW91ckNvbXBvbmVudCA6d2lkdGg9XCI0MlwiIDpoZWlnaHQ9XCI0MlwiLz5cbjwvdGVtcGxhdGU+IiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9zZmMudnVlanMub3JnL3Z1ZS5ydW50aW1lLmVzbS1icm93c2VyLmpzXCJcbiAgfVxufSIsIllvdXJDb21wb25lbnQudnVlIjoiPHRlbXBsYXRlPlxuXHR3aWR0aO+8mnt7d2lkdGh9fSwgaGVpZ2h077yae3toZWlnaHR9fVxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBzZXR1cD5cblxuZGVmaW5lUHJvcHMoe1xuXHR3aWR0aDoge1xuXHRcdHR5cGU6IE51bWJlcixcblx0XHRkZWZhdWx0OiBudWxsLFxuXHR9LFxuXHRoZWlnaHQ6IHtcblx0XHR0eXBlOiBOdW1iZXIsXG5cdFx0ZGVmYXVsdDogbnVsbCxcblx0fSxcbn0pO1xuPC9zY3JpcHQ+In0=).
